packages/ra-ui-materialui/src/list/DatagridHeaderCell.js

```diff
@@ -35,9 +35,12 @@ export const DatagridHeaderCell = ({
                 enterDelay={300}
             >
                 <TableSortLabel
-                    active={field.props.source === currentSort.field}
+                    active={
+                        (field.props.sortBy || field.props.source) ===
```

Comment:
Hard to read in this shape, perhaps you can invert the test to have the `||` in the rightmost part?